pkg/skaffold/jib/jib.go

```diff
@@ -88,6 +88,45 @@ type filesLists struct {
 // watchedFiles maps from project name to watched files
 var watchedFiles = map[string]filesLists{}
 
+// GetDependencies returns a list of files to watch for changes to rebuild
+func GetDependencies(ctx context.Context, workspace string, artifact *latest.JibArtifact) ([]string, error) {
+	t, err := DeterminePluginType(workspace, artifact)
+	if err != nil {
+		return nil, err
+	}
+
+	switch t {
+	case JibMaven:
+		return getDependenciesMaven(ctx, workspace, artifact)
+	case JibGradle:
+		return getDependenciesGradle(ctx, workspace, artifact)
+	default:
+		return nil, errors.Errorf("Unable to determine Jib builder type for %s", workspace)
+	}
+}
+
+// DeterminePluginType tries to determine the Jib plugin type for the given artifact.
+func DeterminePluginType(workspace string, artifact *latest.JibArtifact) (PluginType, error) {
+	// check if explicitly specified
+	if artifact != nil {
+		if t := PluginType(artifact.Type); t.IsKnown() {
+			return t, nil
+		}
+	}
+
+	// check for typical gradle files
+	for _, gradleFile := range []string{"build.gradle", "gradle.properties", "settings.gradle", "gradlew", "gradlew.bat", "gradlew.cmd"} {
+		if util.IsFile(filepath.Join(workspace, gradleFile)) {
+			return JibGradle, nil
+		}
+	}
+	// check for typical maven files; .mvn is a directory used for polyglot maven
+	if util.IsFile(filepath.Join(workspace, "pom.xml")) || util.IsDir(filepath.Join(workspace, ".mvn")) {
+		return JibMaven, nil
+	}
+	return -1, errors.Errorf("Unable to determine Jib plugin type for %s", workspace)
```

Comment:
Didn't you say you would use 0 for unknown?

Reply:
Because PluginType is an integer, its default value is 0 (e.g., when creating a new instance or when unmarshalling a YAML structure), and is effectively means "guess".  Whereas at this point, we have guessed and failed.  So I think something non-zero is better.